File: sales-flow-ai/frontend/src/api/types/goals.ts
/**
 * ╔════════════════════════════════════════════════════════════════════════════╗
 * ║  GOAL API TYPES                                                            ║
 * ║  TypeScript Types für Goal Engine API                                      ║
 * ╚════════════════════════════════════════════════════════════════════════════╝
 */

// ============================================
// ENUMS
// ============================================

export type GoalType = 'income' | 'rank';
export type GoalStatus = 'active' | 'achieved' | 'paused' | 'cancelled';

// ============================================
// CONFIG
// ============================================

export interface DailyFlowConfig {
  working_days_per_week: number;
  contact_to_customer_rate: number;
  contact_to_partner_rate: number;
  followups_per_customer: number;
  followups_per_partner: number;
  reactivation_share: number;
}

export const DEFAULT_DAILY_FLOW_CONFIG: DailyFlowConfig = {
  working_days_per_week: 5,
  contact_to_customer_rate: 0.2,
  contact_to_partner_rate: 0.05,
  followups_per_customer: 3,
  followups_per_partner: 5,
  reactivation_share: 0.2,
};

// ============================================
// REQUEST TYPES
// ============================================

export interface GoalCalculateRequest {
  company_id: string;
  region?: string;
  goal_type: GoalType;
  target_monthly_income?: number;
  target_rank_id?: string;
  timeframe_months: number;
  current_group_volume?: number;
  config?: DailyFlowConfig;
}

export interface GoalSaveRequest {
  company_id: string;
  goal_type: GoalType;
  target_monthly_income?: number;
  target_rank_id?: string;
  target_rank_name?: string;
  timeframe_months: number;
  calculated_group_volume?: number;
  calculated_customers?: number;
  calculated_partners?: number;
}

// ============================================
// RESPONSE TYPES
// ============================================

export interface GoalCalculateResponse {
  success: boolean;
  
  // Target
  target_rank_id: string;
  target_rank_name: string;
  company_id: string;
  company_name: string;
  
  // Volume
  required_group_volume: number;
  missing_group_volume: number;
  
  // Estimates
  estimated_customers: number;
  estimated_partners: number;
  
  // Time distribution
  per_month_volume: number;
  per_week_volume: number;
  per_day_volume: number;
  timeframe_months: number;
  
  // Daily targets
  weekly_new_contacts: number;
  weekly_followups: number;
  weekly_reactivations: number;
  daily_new_contacts: number;
  daily_followups: number;
  daily_reactivations: number;
  
  // Text
  summary_text: string;
  disclaimer: string;
}

export interface GoalSaveResponse {
  success: boolean;
  goal_id?: string;
  message: string;
  error?: string;
}

export interface DailyTargetsResponse {
  has_goal: boolean;
  
  // Goal info
  company_id?: string;
  company_name?: string;
  target_rank_name?: string;
  target_monthly_income?: number;
  
  // Progress
  days_remaining?: number;
  progress_percent?: number;
  
  // Daily targets
  daily_new_contacts: number;
  daily_followups: number;
  daily_reactivations: number;
  
  // Weekly targets
  weekly_new_contacts: number;
  weekly_followups: number;
}

// ============================================
// COMPANY & RANK TYPES
// ============================================

export interface CompanyInfo {
  id: string;
  name: string;
  logo: string;
  region: string;
}

export interface CompanyListResponse {
  companies: CompanyInfo[];
  count: number;
}

export interface RankInfo {
  id: string;
  name: string;
  order: number;
  min_group_volume?: number;
  avg_monthly_income?: number;
  income_range?: [number, number];
}

export interface RankListResponse {
  company_id: string;
  company_name: string;
  ranks: RankInfo[];
  count: number;
}

// ============================================
// GOAL SUMMARY
// ============================================

export interface GoalSummary {
  goal_id: string;
  company_id: string;
  goal_type: GoalType;
  target_monthly_income?: number;
  target_rank_name?: string;
  timeframe_months: number;
  start_date: string;
  end_date: string;
  days_remaining: number;
  progress_percent: number;
  status: GoalStatus;
  
  daily_new_contacts: number;
  daily_followups: number;
  daily_reactivations: number;
}

File: _BACKUP_OLD/sales-flow-ai/backend/app/api/schemas/goals.py
"""
╔════════════════════════════════════════════════════════════════════════════╗
║  GOAL API SCHEMAS                                                          ║
║  Request/Response Models für Goal Endpoints                                ║
╚════════════════════════════════════════════════════════════════════════════╝
"""

from pydantic import BaseModel, Field
from typing import Optional, List
from uuid import UUID
from datetime import date

from app.domain.goals.models import (
    GoalType,
    GoalStatus,
    DailyFlowConfig,
    DailyFlowTargets,
    DEFAULT_DAILY_FLOW_CONFIG,
)
from app.domain.compensation.models import RankDefinition


# ============================================
# CALCULATE GOAL
# ============================================

class GoalCalculateRequest(BaseModel):
    """Request to calculate goal targets."""
    
    company_id: str = Field(..., description="Company identifier (e.g., 'zinzino')")
    region: str = Field(default="DE", description="Region code")
    goal_type: GoalType = Field(..., description="Type of goal: income or rank")
    
    target_monthly_income: Optional[float] = Field(
        None, ge=0, le=100000,
        description="Target monthly income in EUR (for income goals)"
    )
    target_rank_id: Optional[str] = Field(
        None, description="Target rank ID (for rank goals)"
    )
    
    timeframe_months: int = Field(
        default=6, ge=1, le=60,
        description="Timeframe to achieve goal in months"
    )
    current_group_volume: float = Field(
        default=0, ge=0,
        description="Current group volume"
    )
    
    config: Optional[DailyFlowConfig] = Field(
        default=None,
        description="Custom daily flow configuration"
    )
    
    class Config:
        json_schema_extra = {
            "example": {
                "company_id": "zinzino",
                "region": "DE",
                "goal_type": "income",
                "target_monthly_income": 2000,
                "timeframe_months": 6,
                "current_group_volume": 0,
            }
        }


class GoalCalculateResponse(BaseModel):
    """Response with calculated goal targets."""
    
    success: bool = True
    
    # Target
    target_rank_id: str
    target_rank_name: str
    company_id: str
    company_name: str
    
    # Volume
    required_group_volume: float
    missing_group_volume: float
    
    # Estimates
    estimated_customers: int
    estimated_partners: int
    
    # Time distribution
    per_month_volume: float
    per_week_volume: float
    per_day_volume: float
    timeframe_months: int
    
    # Daily targets
    weekly_new_contacts: float
    weekly_followups: float
    weekly_reactivations: float
    daily_new_contacts: float
    daily_followups: float
    daily_reactivations: float
    
    # For frontend display
    summary_text: str
    disclaimer: str


# ============================================
# SAVE GOAL
# ============================================

class GoalSaveRequest(BaseModel):
    """Request to save a user goal."""
    
    company_id: str
    goal_type: GoalType
    target_monthly_income: Optional[float] = None
    target_rank_id: Optional[str] = None
    target_rank_name: Optional[str] = None
    timeframe_months: int = Field(default=6, ge=1, le=60)
    
    # Calculated values
    calculated_group_volume: Optional[float] = None
    calculated_customers: Optional[int] = None
    calculated_partners: Optional[int] = None
    
    # Daily targets to save
    daily_targets: Optional[DailyFlowTargets] = None
    config: Optional[DailyFlowConfig] = None


class GoalSaveResponse(BaseModel):
    """Response after saving a goal."""
    
    success: bool
    goal_id: Optional[UUID] = None
    message: str
    error: Optional[str] = None


# ============================================
# GET DAILY TARGETS
# ============================================

class DailyTargetsResponse(BaseModel):
    """Response with user's current daily targets."""
    
    has_goal: bool
    
    # Goal info (if exists)
    company_id: Optional[str] = None
    company_name: Optional[str] = None
    target_rank_name: Optional[str] = None
    target_monthly_income: Optional[float] = None
    
    # Progress
    days_remaining: Optional[int] = None
    progress_percent: Optional[float] = None
    
    # Daily targets
    daily_new_contacts: float = 0
    daily_followups: float = 0
    daily_reactivations: float = 0
    
    # Weekly targets
    weekly_new_contacts: float = 0
    weekly_followups: float = 0


# ============================================
# GOAL LIST
# ============================================

class GoalSummary(BaseModel):
    """Summary of a user goal."""
    
    goal_id: UUID
    company_id: str
    goal_type: GoalType
    target_monthly_income: Optional[float]
    target_rank_name: Optional[str]
    timeframe_months: int
    start_date: date
    end_date: date
    days_remaining: int
    progress_percent: float
    status: GoalStatus
    
    daily_new_contacts: float
    daily_followups: float
    daily_reactivations: float


class GoalListResponse(BaseModel):
    """Response with list of user goals."""
    
    goals: List[GoalSummary]
    active_goal: Optional[GoalSummary] = None

